### PR TITLE
remove pull secret copy, defer to image registry

### PR DESCRIPTION
**Pull request by gabemontero**
With @ricardomaraschini 's https://github.com/openshift/image-registry/pull/226 merged here is my change here the change that remove's samples operator's copying of the install pull secret to the openshift namespace

I still have my prometheus metrics around whether the pull secret has TBR credentials, but it now just looks at the install pull secret directly.

Have not tried it in a local cluster yet so marked WIP for now

@openshift/openshift-team-developer-experience FYI

### CODE REVIEW

**Comment by gabemontero** [comment]
@ricardomaraschini ... the image-ecosystem failures are around imagestream imports failing because of credentials.

In some of the test output I seen message like `Mar 18 14:08:37.236: INFO: after seeing FailedImageImports for ruby a manual image-import was submitted`

And when I examine the imagestreams in the must gather located at https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_cluster-samples-operator/249/pull-ci-openshift-cluster-samples-operator-master-e2e-aws-image-ecosystem/922/artifacts/e2e-aws-image-ecosystem/must-gather.tar

I see consistently see this across all the imagestreams:

```
        message: 'Internal error occurred: registry.redhat.io/3scale-amp23/apicast-gateway:latest:
          Get https://registry.redhat.io/v2/3scale-amp23/apicast-gateway/manifests/latest:
          unauthorized: Please login to the Red Hat Registry using your Customer Portal
          credentials. Further instructions can be found here: https://access.redhat.com/RegistryAuthentication'
```

I would contend your image-registry change is not working as expected.

**Comment by gabemontero** [comment]
in talking with @ricardomaraschini it looks like this PR would need https://github.com/openshift/openshift-apiserver/pull/83 and https://github.com/openshift/cluster-openshift-apiserver-operator/pull/284 as well

File: pkg/operator/controller.go [hunk 1]
```diff
@@ -3,17 +3,16 @@ package operator
 import (
 	"fmt"
 	"github.com/openshift/cluster-samples-operator/pkg/util"
+	kubeset "k8s.io/client-go/kubernetes"
```
Comment (gabemontero) [inline pkg/operator/controller.go:6]: I'll reorder this once e2e's pass and we move forward on this

File: pkg/util/util.go [hunk 2]
```diff
@@ -299,33 +294,7 @@ func ClusterOperatorStatusProgressingCondition(s *samplev1.Config, degradedState
...
-			foundImportCred = true
-			break
-		}
-	}
-	if !foundImportCred {
-		return true
-	}
-
-	return ConditionFalse(s, samplev1.ImportCredentialsExist)
+	return false
```
Comment (gabemontero) [inline pkg/util/util.go:297]: could just delete this method as well ... will also do when we have e2e's passing

File: pkg/stub/handler.go [hunk 3]
```diff
@@ -487,7 +477,15 @@ func (h *Handler) CreateDefaultResourceIfNeeded(cfg *v1.Config) (*v1.Config, err
 func (h *Handler) initConditions(cfg *v1.Config) *v1.Config {
 	now := kapis.Now()
 	util.Condition(cfg, v1.SamplesExist)
-	util.Condition(cfg, v1.ImportCredentialsExist)
+	creds := util.Condition(cfg, v1.ImportCredentialsExist)
```
Comment (gabemontero) [inline pkg/stub/handler.go:480]: not deleting this condition in case customers are coding to it

could entertain marking deprecated in openshift/api

**Comment by gabemontero** [comment]
ci build pain wit its internal imagestreams with e2e-aws and e2e-aws-upgrade

we'll see what happens with the other 2 in flight e2es

**Comment by gabemontero** [comment]
yeah @dmage @ricardomaraschini I'm still getting `unauthorized: Please login to the Red Hat Registry using your Customer Portal credentials. ` errors on the sample imagestreams if the samples operator does not copy the install pull secret

**Comment by ricardomaraschini** [comment]
> yeah @dmage @ricardomaraschini I'm still getting `unauthorized: Please login to the Red Hat Registry using your Customer Portal credentials. ` errors on the sample imagestreams if the samples operator does not copy the install pull secret

Very interesting. How does the update work on CI? I mean, how can we be sure that we did have the new version of the apiserver when this test happened?

**Comment by ricardomaraschini** [comment]
Yeah, it seems like it is running the new version(judging by the artifacts). No idea why it is not working at this stage.

**Comment by gabemontero** [comment]
> Yeah, it seems like it is running the new version(judging by the artifacts). No idea why it is not working at this stage.

@ricardomaraschini @dmage perhaps it makes sense when you guys have cycles to use cluster bot on slack to launch a cluster from this PR and start investigating.

**Comment by ricardomaraschini** [comment]
/test e2e-aws-operator

@gabemontero @dmage I ran e2e tests using this PR branch on a cluster and they succeeded. I am not sure if these tests use the hypershift `sidecar` and if that might be causing issue(I don't think we are mounting the node credentials there).

```
[rmarasch@loaner ~]$ oc version
Client Version: 4.5.0-0.nightly-2020-04-10-040610
Server Version: 4.5.0-0.nightly-2020-04-10-040610
Kubernetes Version: v1.18.0-rc.1
[rmarasch@loaner ~]$
```

**Comment by ricardomaraschini** [comment]
Seems like the failure has changed now for `e2e-aws-operator`.

**Comment by gabemontero** [comment]
updated this PR with @ricardomaraschini 's https://github.com/gabemontero/cluster-samples-operator/pull/2

thanks @ricardomaraschini

**Comment by gabemontero** [comment]
terraform / aws flake with latest e2e-aws-operator

/test e2e-aws-operator

**Comment by gabemontero** [comment]
failed to acquire lease e2e upgrade

/test e2e-aws-upgrade

**Comment by gabemontero** [comment]
failed to acquire lease image eco

/test e2e-aws-image-ecosystem

**Comment by gabemontero** [comment]
ok @dmage @ricardomaraschini we got imagestream imports in the openshift namespace working via the latest image-eco and operator e2e's !!

based on how CI health is on Friday, we'll see about getting this in

**Comment by gabemontero** [comment]
the latest e2e-aws at least go installed and had passing tests when I looked a few mintutes ago

**Pull request merged**